CMakeLists.txt

```diff
@@ -220,7 +220,7 @@ if(BML_OPENCL)
   include(FindOpenCL)
 endif()
 
-SET(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
+SET(CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
```

Comment:
Why change those to `PROJECT_SOURCE_DIR`? That's semantically different than using `CMAKE_SOURCE_DIR`. In our case it ends up being the same, but if we were ever to move the `project()` call to another `CMakeLists.txt` this would break.

Reply:
We are trying to do a `LATTE` super-build with:
```cmake
cmake_minimum_required(VERSION 3.9)
project(latte_super)

add_subdirectory(bml)
add_subdirectory(qmd-progress)
option(PROGRESS "Use progress library" ON)
add_subdirectory(latte/cmake)
```
and if you include `bml` like this `CMAKE_SOURCE_DIR` points to wrong directory, while `PROJECT_SOURCE_DIR` is correct.

Reply:
The only place where I wasn't sure is in `test-getter`, but there we could link against `BML::bml` as well.

Reply:
Just a comment that in general the CMake seems to use keywords that do not correspond to the system variables which forces stuff like
export CUDA_TOOLKIT_ROOT_DIR=${CUDADIR}
Is there a reason not to use the env variables directly?

Reply:
bml's cmake build system needs a deep scrub, since cmake-3.8 cuda is a first class citizen, I haven't really touched that part in this PR, but the `CUDA_TOOLKIT_ROOT_DIR` environment variable has always been CMake's variable to detect cuda.

Reply:
OK, I generally find the CMake build system hard to follow at the best of times. It would help a lot if our example build scripts were even close to a working example.

Reply:
The bml is a dependency of LATTE. I would expect that LATTE adjusts to the bml and not the other way around. I am not saying that I see any serious reason to move the `project()` call, but it does feel wrong to make that change here.

Reply:
@nicolasbock well this applies to any time, we want to use `add_subdirectory(bml)` for any reason, latte is just one example. It seems to me most CMake projects have converged to allow that the inclusion for their projects as subdirectories these days, so I followed that.

Reply:
Could you then at least add a comment so that someone does not refactor the `project()` call into a subdirectory?

Reply:
Sorry, but why would somebody try to move the `project()` line?

Reply:
Because you can. The comment is just insurance so that in case a year down the road someone goes wild with refactoring the CMake scripts we remember that we should not move that call.

Reply:
Well, if I move a random line of CMake code into a different file, it will most likely break something, and the `project()` line is no exception.

Reply:
> PROJECT_SOURCE_DIR
Top level source directory for the current project.
This is the source directory of the most recent project() command.

https://cmake.org/cmake/help/latest/variable/PROJECT_SOURCE_DIR.html

However, that's incorrect because scoping in CMake works only down, not up (which I couldn't find good documentation for) and a `project()` call in `add_subdirectory()` does not propagate back to the calling scope.

:+1: on the change.